servo/connectors/kubernetes.py

```diff
@@ -2524,11 +3109,11 @@ class CanaryOptimization(BaseOptimization):
     """
 
     # The deployment and container stanzas from the configuration
-    deployment_config: "DeploymentConfiguration"
+    deployment_config: Union["DeploymentConfiguration", "RolloutConfiguration"]
     container_config: "ContainerConfiguration"
 
     # State for mainline resources. Read from the cluster
-    deployment: Deployment
+    deployment: Union[Deployment, Rollout]
```

Comment:
This is a bad idea. Add a `rollout` field and then use a validator to ensure that you can configure a `deployment` or a `rollout` but not both. Overloading the `deployment` key like this is no bueno